src/luarocks/cmd/nix.lua

```diff
@@ -90,7 +119,16 @@ end
 -- in which case it uses the special nixpkgs uris mirror://luarocks
 local function gen_src_from_basic_url(url)
    assert(type(url) == "string")
-   local checksum = get_basic_checksum(url)
+
+   local command = "nix-prefetch-url "..url
+   local fetcher = "fetchurl"
+
+   local checksum, unpack = get_basic_checksum(command)
+   if unpack then
+      command = "nix-prefetch-url --unpack "..url
```

Comment:
this command is never run and we find ourselves with a wrong hash. For instance I've tried `luarocks nix luaevent` and it used fetchTarball but with the fetchurl hash and thus failed to build.

Reply:
Good catch